src/main/java/org/minions/devfund/bruno/battleship/Ship.java

```diff
@@ -0,0 +1,265 @@
+     * @param column integer value.
+     * @return boolean value.
+     */
+    public boolean wasShootAt(final int row, final int column) {
+        return horizontal ? hit[column - bowColumn] : hit[row - bowRow];
+    }
+
+    /**
+     * If a part of the ship occupies the given row and column, and the ship hasn't been sunk mark
+     * a part of the ship as 'hit'.
+     *
+     * @param row    integer value.
+     * @param column integer value.
+     * @return boolean value.
+     */
+    public boolean shootAt(final int row, final int column) {
+        boolean rowBowEquals = row == bowRow;
+        boolean columnsBowEquals = column == bowColumn;
+        int bowColumnLength = bowColumn + length;
+        int bowRowLength = bowRow + length;
+        int placeHitColumn = column - bowColumn;
+        int placeHitRow = row - bowRow;
+        if (!isSunk()) {
+            return horizontal ? isBowHit(column, rowBowEquals, bowColumnLength, placeHitColumn)
+                    : isBowHit(row, columnsBowEquals, bowRowLength, placeHitRow);
+        }
+        return false;
+    }
+
+    /**
+     * Marks a part of the ship as 'hit' (in the hit array, 0 indicates the bow) and return true,
+     * otherwise return false.
+     *
+     * @param columnRow integer value.
+     * @param bowEquals boolean value.
+     * @param length    integer length.
+     * @param place     integer hit.
+     * @return true if in the hit array, 0 indicates the bow, otherwise return false
+     */
+    public boolean isBowHit(final int columnRow, final boolean bowEquals, final int length, final int place) {
+        if (bowEquals && columnRow < length) {
+            hit[place] = true;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if every part of the ship has been hit, false otherwise.
+     *
+     * @return boolean is sunk value.
+     */
+    public boolean isSunk() {
+        for (boolean place : hit) {
+            if (!place) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Return 'x' if the ship has been sunk, 'S' if it has not been sunk.
+     * This method can be used to print out locations in the ocean that have been shot at;
+     * it should not be used to print locations that have not been shot at.
+     *
+     * @return String value.
+     */
+    @Override
+    public String toString() {
+        return isSunk() ? "x" : "S";
+    }
+
+    /**
+     * Gets the ship length.
+     *
+     * @return integer length value.
+     */
+    public int getLength() {
+        return length;
+    }
+
+    /**
+     * Gets the ship length.
+     *
+     * @param length integer value.
+     */
+    public void setLength(final int length) {
+        this.length = length;
+    }
+
+    /**
+     * Gets the ship hits.
+     *
+     * @return boolean array hit.
+     */
+    public boolean[] getHit() {
+        boolean[] newHit = this.hit;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "newHit". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)